HabitTracker.alvaromosconi/HabitTracker.alvaromosconi/HabitRepository.cs

```diff
@@ -0,0 +1,405 @@
+        {
+            connection.Open();
+
+            var command = connection.CreateCommand();
+            command.CommandText =
+                $@"
+                            SELECT * FROM {TABLE_NAME}
+                        ";
+
+            SqliteDataReader reader = command.ExecuteReader();
+
+            while (reader.Read())
+            {
+                tableData.Add(
+                    new Habit
+                    {
+                        Id = reader.GetInt32(0),
+                        Name = reader.GetString(1),
+                        Date = DateTime.ParseExact(reader.GetString(2), "dd-MM-yy", new CultureInfo("en-US")),
+                        Quantity = reader.GetInt32(3),
+                    });
+            }
+
+            connection.Close();
+        }
+
+    }
+
+    internal void ExecuteOption(string userInput)
+    {
+        switch (userInput)
+        {
+            case "1":
+                PrintAllRecords();
+                break;
+            case "2":
+                InsertNewRecord();
+                break;
+            case "3":
+                UpdateAnExistingRecord();
+                break;
+            case "4":
+                DeleteAnExistingRecord();
+                break;
+        }
+
+        PrintBackOption();
+    }
+
+    private void PrintBackOption()
+    {
+
+        Console.WriteLine("\nPress any key to continue");
+        Console.ReadKey();
+    }
+
+    private void PrintAllRecords()
+    {
+
+        var groupedHabits = tableData.GroupBy(habit => habit.Name);
+
+        foreach (var group in groupedHabits)
+        {
+            Console.ForegroundColor = Console.ForegroundColor = ConsoleColor.DarkBlue; 
+            Console.WriteLine($"\nGroup: {group.Key}\n");
+            Console.ResetColor();
+            Console.WriteLine("\nId     Date        Quantity");
+            Console.WriteLine("---------------------------");
+            foreach (Habit habit in group)
+            {
+                Console.WriteLine($"{habit.Id,-4} {habit.Date,-15:dd-MM-yy} {habit.Quantity}");
+            }
+
+            Console.WriteLine();
+        }
+    }
+
+
+    internal void InsertNewRecord()
+    {
+        List<string> existingHabitNames = GetExistingHabitNames();
+
+        if (existingHabitNames.Count == 0)
+        {
+            Insert(null);
+        }
+        else
+        {
+            Console.WriteLine("Please type:");
+            Console.WriteLine("\n 0. To back. \n 1. To create a new habit. \n 2. To register a new record in an existing habit.");
+            string userChoice = Console.ReadLine();
+            switch (userChoice)
+            {
+                case "0":
+                    break;
+                case "1":
+                    Console.Clear();
+                    Insert(null);
+                    break;
+                case "2":
+                    Console.Clear();
+                    PrintAllExistingHabitNames(existingHabitNames);
```

Comment:
Could you please fix this issue? 😊 Also this is optional but I recommend you create a method when a case in the switch statement gets too bloated. You want your switch statements to be lean and easy to read. 

![image](https://github.com/TheCSharpAcademy/CodeReviews.Console.HabitTracker/assets/108711476/32801706-e991-4aa7-9fb6-3f700556decf)

Reply:
@alvaromosconi Let us know if you need help 😁

Reply:
Thanks for offering help, I think the app is now fine. Sorry for not having responded. I've been busy with other things these days :/

I am also making the Flashcards app. But I think I'm going to have to do this one again because when I was about to incorporate the study session part I realized that I might not make the best design decisions. Would you mind taking a look at it? it would help me a lot. 

This is the repo: https://github.com/alvaromosconi/CodeReviews.Console.Flashcards